### DISCUSSION TITLE
Sorting with global filter

### DISCUSSION THREAD

**Discussion by DenizUgur** [TanStack/table#5025] [Q&A]
What am I missing to sort the rows based on global filter? I would like to integrate Fuse.js to do weighted search but whatever I do feels hacky. What is the recommended way of achieving this?

If I use `globalFilterFn` I can filter based on the `refIndex` (returned by Fuse.js) and I can then add `score` to `addMeta`. But then how do I use that meta information to sort the rows?

**Comment by EliteUser** [comment]
Faced the same problem. I want to sort (without using sorting state) the results of global filtering by `itemRank` in `meta`, but I don't understand how to do it. I thought that `table.getSortedRowModel()` can help, but rows inside it don't have any meta info
